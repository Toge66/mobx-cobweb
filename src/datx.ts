/***************************************************
 * Created by nanyuantingfeng on 2020/7/28 09:28. *
 ***************************************************/
export * from '@issues-beta/datx'
