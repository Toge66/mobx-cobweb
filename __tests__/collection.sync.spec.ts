import { collection, useFixturesByGET } from './config'
import { modelToJSON } from 'datx'
import Department from './models/Department'
// tslint:disable-next-line:no-var-requires
const path = require('path')

describe('collection.sync', () => {
  let scope: any = null

  beforeAll(() => {
    scope = useFixturesByGET()
  })

  beforeEach(() => {
    collection.removeAll(Department)
  })

  const cases = (sf: Department[]) => {
    expect(sf[0].children[0].parentId === sf[0]).toBeTruthy()
    expect(collection.findAll(Department).length).toBe(2035)

    const dd = collection.findOne<Department>(Department, '6Rk9l1WYNM0400:50128173')
    expect(dd.parentId.id).toBe('6Rk9l1WYNM0400:1')
    expect(dd.name).toBe('研发总部')
    expect(dd.active).toBe(true)
    expect(modelToJSON(dd)).toMatchSnapshot()
  }

  const data: object[] = require(path.join(__dirname, './fixtures/organization.departments.json')).items

  test('should sync IRawData', () => {
    const sf = collection.sync<Department>({
      type: 'organization.Department',
      data
    })
    cases(sf)
  })

  test('should sync data & type', () => {
    const sf = collection.sync<Department>(Department, data)
    cases(sf)
  })

  test('should sync data & type:string ', () => {
    const sf = collection.sync<Department>('organization.Department', data)
    cases(sf)
  })
})
