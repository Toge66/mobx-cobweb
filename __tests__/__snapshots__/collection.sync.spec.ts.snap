// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection.sync by list-data should sync IRawData 1`] = `
Object {
  "__META__": Object {
    "MODEL_PERSISTED_FIELD": true,
    "collection": undefined,
    "fields": Object {
      "children": Object {
        "defaultValue": undefined,
        "referenceDef": Object {
          "model": "organization.Department",
          "property": undefined,
          "type": 1,
        },
      },
      "code": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "id": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "name": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "parent": Object {
        "defaultValue": undefined,
        "referenceDef": Object {
          "model": "organization.Department",
          "type": 0,
        },
      },
      "parentId": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
    },
    "id": "875fdcddd47e",
    "type": "organization.Department",
  },
  "children": Array [
    Object {
      "id": "bd3e51d15c92",
      "type": "organization.Department",
    },
  ],
  "code": 3043,
  "id": "875fdcddd47e",
  "name": "Bedfordshire",
  "parent": Object {
    "id": "66f58b59c384",
    "type": "organization.Department",
  },
  "parentId": "66f58b59c384",
}
`;

exports[`collection.sync by tree-data should sync data & type 1`] = `
Object {
  "__META__": Object {
    "collection": undefined,
    "fields": Object {
      "children": Object {
        "defaultValue": undefined,
        "referenceDef": Object {
          "model": "organization.Department",
          "property": undefined,
          "type": 1,
        },
      },
      "code": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "id": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "name": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
      "parent": Object {
        "defaultValue": undefined,
        "referenceDef": Object {
          "model": "organization.Department",
          "type": 0,
        },
      },
      "parentId": Object {
        "defaultValue": undefined,
        "referenceDef": false,
      },
    },
    "id": "875fdcddd47e",
    "type": "organization.Department",
  },
  "children": Array [
    Object {
      "id": "f66bb6bc1c3c",
      "type": "organization.Department",
    },
    Object {
      "id": "1511422581a9",
      "type": "organization.Department",
    },
    Object {
      "id": "00cf0fa95834",
      "type": "organization.Department",
    },
    Object {
      "id": "bd3e51d15c92",
      "type": "organization.Department",
    },
  ],
  "code": 31517,
  "id": "875fdcddd47e",
  "name": "ADP",
  "parent": Object {
    "id": "66f58b59c384",
    "type": "organization.Department",
  },
  "parentId": "66f58b59c384",
}
`;
